x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/components/processor_form/field_components/xjson_and_json_editor.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { XJsonLang } from '@kbn/monaco';
+import React, { FunctionComponent, useCallback } from 'react';
+import { FieldHook } from '../../../../../../shared_imports';
+
+import { TextEditor } from './text_editor';
+
+interface Props {
+  field: FieldHook<string>;
+  editorProps: { [key: string]: any };
+}
+
+const defaultEditorOptions = {
+  minimap: { enabled: false },
+  lineNumbers: 'off',
+};
+
+export const XJsonAndJsonEditor: FunctionComponent<Props> = ({ field, editorProps }) => {
```

Comment:
We could simply name it `XJsonEditor` since json language is a subset of xjson, so if an editor is xjson, it means that it should also support json.